### ISSUE TITLE
make description a required attribute in the module.xml

### ISSUE THREAD

**Issue by SergeyMi37** [intersystems/ipm#210]

**Comment by isc-tleavitt** [comment]
This would be a major backward incompatibility. It would be better to drive users to better habits through tools.

**Comment by isc-kiyer** [comment]
@isc-tleavitt I agree we shouldn't make it required during load but perhaps could make it required for the publish command to force the best practice? That would keep compatibility for downloading existing packages but force any new modules/versions published to add a version

**Comment by evshvarov** [comment]
I second @isc-kiyer suggestion.

**Comment by isc-tleavitt** [comment]
Fair points. Gut feeling: restrictions on what can be published should be under control of the registry itself, not the package manager. The registry could theoretically inject further constraints about naming conventions, package signing (eventually), artifact formats, etc. that could differ between registries.

@evshvarov how would this interact with the current automatic publishing that happens on OEX? Will people get stuck with "released" but not-actually-releasable versions?

Reopening to continue the conversation...

**Comment by evshvarov** [comment]
Yes, I agree it is more registry's feature than IPM client.
